### ISSUE TITLE
Token refresh confusion

### ISSUE THREAD

**Issue by brainfuel** [PayPal-Mobile/ios-here-sdk-dist#76]
I'm a little confused by token refresh. 

Currently I'm getting the composite token fine, I call setupWithCompositeTokenString, everything is good, I can make transactions etc.
However 8 hours later it expires, and at that point I'm confused.

The documentation seems to imply that refresh is taken care of automatically by the SDK "When the access token expires,  we will automatically invoke this endpoint to refresh the access token"

But I still get a 401, so it seems it's not taken care of 'automatically'.

Could you clarify what needs to be done after setupWithCompositeTokenString to refresh the token lease?

**Comment by djMax** [comment]
It's not "automatic" in that you still have to provide a server that will refresh. Check out this long explanation: https://github.com/paypal/paypal-retail-node

TL;DR - when the token expires the SDK will call the refreshUrl and expect to get a new token back. If it doesn't, or if that token is also expired, it will fail with a 401.

**Comment by brainfuel** [comment]
Thanks for your reply. I'm using heroku, and I am sent the refresh_url.  But when it expires it is not refreshed by the SDK.
Currently I'm having to perform the GET request on the refresh_url manually to get around this, but I assume from the documentation this is not how the SDK is supposed to work.

**Comment by brainfuel** [comment]
Manual refresh workaround seems to work OK. Closing
